internal/store/serverservice.go

```diff
@@ -179,6 +186,44 @@ func (s *Serverservice) Create(ctx context.Context, serverID uuid.UUID, createCo
 	return err
 }
 
+// AddServer creates a server record in FleetDB
+// @id: required
+// @ip: required
+// @user: required
+// @pwd: required
+func AddServer(ctx context.Context, s *Serverservice, serverID uuid.UUID, facilityCode, bmcAddr, bmcUser, bmcPass string) error {
+	if bmcUser == "" {
+		return ErrEmptyBmcUser
+	}
+
+	addr, err := netip.ParseAddr(bmcAddr)
+	if err != nil {
+		return err
+	}
+
+	// Add server
+	server := sservice.Server{UUID: serverID, Name: serverID.String(), FacilityCode: facilityCode}
+	_, _, err = s.client.Create(ctx, server)
+	if err != nil {
+		if strings.Contains(err.Error(), serverExistErrMsg) {
```

Comment:
Oh I was following the other comment: https://github.com/metal-toolbox/conditionorc/pull/132#discussion_r1376050796

Basically `Check if the error returned by HSS/Fleetdb contains a duplicate key error or similar (which indicates the server already exists) then return http.StatusBadRequest, for other errors return http.StatusInternalServerError`.

I don't have preference on it since `mtcl` will see the same error message which users can tell if the id is duplicated or not. Maybe `BadRequest` is appropriate?

I don't have a good way besides reading the error message. I don't find useful information inside `ServerResponse`